### PR TITLE
MM-16997: Simplified TestUploadFiles

### PR DESCRIPTION
**Pull request by levb**
#### Summary
Simplified file upload tests
- The tests now fully buffer the data before uploading it. The prior
  code was much more complex because it was intended to eventually
  mature into generic client code; not a priority, definitely not
  valuable for this test.
- Reduced the test stream size from 50Mb to 1Mb.
- Restored contents checks that were removed after a go upgrade 
  changed the generated binary images.
- Also improved error handling in TestHookFileWillBeUploaded for
  MM-16760.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-16997
~~https://mattermost.atlassian.net/browse/MM-16760~~ did not provide new info

### CODE REVIEW

File: app/plugin_hooks_test.go [hunk 1]
```diff
@@ -616,13 +621,17 @@ func TestHookFileWillBeUploaded(t *testing.T) {
 			}
 
 			func (p *MyPlugin) FileWillBeUploaded(c *plugin.Context, info *model.FileInfo, file io.Reader, output io.Writer) (*model.FileInfo, string) {
-				p.API.LogDebug(info.Name)
 				var buf bytes.Buffer
-				buf.ReadFrom(file)
-				p.API.LogDebug(buf.String())
+				n, err := buf.ReadFrom(file)
+				if err != nil {
+					return info, fmt.Sprintf("FAILED to read input fileL n: %v, err: %v", n, err)
```
Comment (lieut-data) [inline app/plugin_hooks_test.go]: Is `fileL` intentional here?

File: app/plugin_hooks_test.go [hunk 2]
```diff
@@ -616,13 +621,17 @@ func TestHookFileWillBeUploaded(t *testing.T) {
...
+				}
 
 				outbuf := bytes.NewBufferString("changedtext")
-				io.Copy(output, outbuf)
+				n, err = io.Copy(output, outbuf)
+				if int(n) != len("changedtext") || err != nil {
+					return info, fmt.Sprintf("FAILED to write output fileL n: %v, err: %v", n, err)
```
Comment (lieut-data) [inline app/plugin_hooks_test.go]: Is `fileL` intentional here?

**Comment by levb** [comment]
Interestingly, this PR failed on the very `TestHookFileWillBeUploaded/updated` PR that it was supposed to provide more information on, and it didn't - there were no errors. 

Also, the test checks the fileinfo in https://github.com/mattermost/mattermost-server/blob/aac4b4e6b8957e1004de6b7e2fd4871ed74a258b/app/plugin_hooks_test.go#L663 before it fails checking for modified file contents in line 669 below.

```
        testing.go:20: {"level":"debug","ts":1564070706.163045,"caller":"plugin/hclog_adapter.go:33","msg":"plugin address","plugin_id":"jc65bxakhfg4i8ibuyh7g6kf7a","wrapped_extras":"address/tmp/plugin268279216networkunixtimestamp2019-07-25T16:05:06.159Z"}
        plugin_hooks_test.go:669: 
            	Error Trace:	plugin_hooks_test.go:669
            	Error:      	Not equal: 
            	            	expected: "changedtext"
            	            	actual  : "inputfile"
            	            	
            	            	Diff:
            	            	--- Expected
            	            	+++ Actual
            	            	@@ -1 +1 @@
            	            	-changedtext
            	            	+inputfile
            	Test:       	TestHookFileWillBeUploaded/updated
```

I am reopening https://mattermost.atlassian.net/browse/MM-16760, but the changes in this PR for `TestHookFileWillBeUploaded` should still go in.

**Comment by lieut-data** [comment]
@levb, the failures you noticed are, I think, related to https://mattermost.atlassian.net/browse/MM-16479 and the ongoing investigation there. Once that goes in, we will hopefully have a bit more detail on which to continue the investigation in that ticket. I think we can still close your https://mattermost.atlassian.net/browse/MM-16760 as such?

**Pull request merged**